projects/packages/plugin-deactivation/src/class-deactivation-handler.php

```diff
@@ -0,0 +1,109 @@
+<?php
+/**
+ * Intercept deactivation of plugins.
+ *
+ * @package automattic/jetpack-plugin-deactivation
+ */
+
+namespace Automattic\Jetpack\Plugin_Deactivation;
+
+/**
+ * Handles plugin deactivation.
+ *
+ * Instantiates the deactivation handler to intercept the deactivation of plugins.
+ */
+class Deactivation_Handler {
+
+	/**
+	 * Plugin Deactivation package version
+	 *
+	 * @var string
+	 */
+	const PACKAGE_VERSION = '0.1.0-alpha';
+
+	/**
+	 * Slug of the plugin to intercept deactivation for.
+	 *
+	 * @var string
+	 */
+	private $plugin;
+
+	/**
+	 * Feedback form url to redicret to, on deactivation.
+	 *
+	 * @var string
+	 */
+	private $feedback_url;
+
+	/**
+	 * Path to a PHP file that will be used as a template for the deactivation dialog.
+	 *
+	 * Copy ./dialog-template.php to your plugin and modify it to suit your needs.
+	 *
+	 * @var string
+	 */
+	private $dialog_view;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param string $plugin       Slug of the plugin to intercept deactivation for.
+	 * @param string $feedback_url Feedback form url to redicret to, on deactivation.
+	 * @param string $dialog_view  Path to a PHP file that will be used as a template for the deactivation dialog.
+	 */
+	public function __construct( $plugin, $feedback_url, $dialog_view ) {
+		$this->plugin       = $plugin;
+		$this->feedback_url = $feedback_url;
+		$this->dialog_view  = $dialog_view;
+
+		add_action( 'load-plugins.php', array( $this, 'enqueue_script' ) );
+		add_action( 'admin_footer-plugins.php', array( $this, 'embed_dialog' ) );
+	}
+
+	/**
+	 * Instantiates the deactivation handler to intercept the deactivation of plugins.
+	 *
+	 * @param string $plugin       Slug of the plugin to intercept deactivation for.
+	 * @param string $feedback_url Feedback form url to redicret to, on deactivation.
+	 * @param string $dialog_view  Path to a PHP file that will be used as a template for the deactivation dialog.
+	 */
+	public static function init( $plugin, $feedback_url, $dialog_view ) {
+		$instance = new self( $plugin, $feedback_url, $dialog_view );
+		return $instance;
+	}
+
+	/**
+	 * Enqueues the deactivation handler script and styles.
+	 */
+	public function enqueue_script() {
+		wp_enqueue_script(
+			'jp-plugin-deactivation',
+			plugins_url( '../dist/deactivation.js', __FILE__ ),
+			array(),
+			self::PACKAGE_VERSION,
+			true
+		);
+
+		wp_enqueue_style(
+			'jp-plugin-deactivation',
+			plugins_url( '../dist/deactivation.css', __FILE__ ),
+			array(),
+			self::PACKAGE_VERSION
+		);
+	}
+
+	/**
+	 * Add required html to the plugin page footer.
+	 */
+	public function embed_dialog() {
+		?>
+		<div id="jp-plugin-deactivation-<?php echo esc_attr( $this->plugin ); ?>" class="jp-plugin-deactivation">
+			<div class="jp-plugin-deactivation__dialog">
+				<?php include $this->dialog_view; ?>
```

Comment:
What type of validation? Throw an error? How about changing it to a `require`?

Reply:
`require` is going to break at runtime, meaning it would be more difficult to detect - breaking only on deactivation, and thus - easier to miss.

I'd recommend checking that the file exists when the class is initialized and if the intention is for this to be a fatal error, fatal upon class construction - that way the developer consuming the class is immediately made aware as opposed to sometime later down the road in a specific condition.

Reply:
I have added a wp_die on `init`